### PR TITLE
Store checkout states in local storage

### PR DESCRIPTION
**Pull request by minkyngkm**
## Done

- Store checkout states in local storage 
- Remove stored local storage after purchasing 

## QA

- Go to `/pro/subscribe`
- Select items
- Refresh page or log in and see the selected items remains as selected.

## Issue / Card

Fixes https://warthogs.atlassian.net/jira/software/c/projects/WD/boards/801?modal=detail&selectedIssue=WD-304

ezgif com-gif-maker (18)

### CODE REVIEW

File: static/js/src/PurchaseModal/components/BuyButton/BuyButton.tsx [hunk 1]
```diff
@@ -87,6 +87,7 @@ const BuyButton = ({ setError, quantity, product, action }: Props) => {
         onSuccess: (data) => {
           //start polling
           setPendingPurchaseID(data);
+          localStorage.clear();
```
Comment (ClementChaumel) [inline static/js/src/PurchaseModal/components/BuyButton/BuyButton.tsx]: Maybe we should just clear pro related values ? so we don't mess up with whoever else could be using the local storage on ubuntu.com

**Review by ClementChaumel — APPROVED** [review]
As mentioned I would make sure to only remove pro-related local storage items. One way to make it easier would be to group them all under a single object in local storage like "pro-product-selector-state" or something

**Comment by minkyngkm** [comment]
> As mentioned I would make sure to only remove pro-related local storage items. One way to make it easier would be to group them all under a single object in local storage like "pro-product-selector-state" or something

Thanks @ClementChaumel for reviewing! I prefixed to the keys and it only removes the prefixed items in the local storage. I didn't group them as a single object though as each value needs to be added in local storage when selected.

**Comment by minkyngkm** [comment]
A new PR has been created at https://github.com/canonical/ubuntu.com/pull/12279

**Pull request closed without merging**